utils/tfw_logger.cpp

```diff
@@ -0,0 +1,214 @@
+#include "mmap_buffer.h"
+#include "clickhouse.h"
+#include "../fw/access_log.h"
+#include <iostream>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <clickhouse/client.h>
+#include <clickhouse/base/socket.h>
+
+using namespace std;
+
+#define FILE_PATH	"/dev/tempesta_mmap_log"
+#define TABLE_NAME	"access_log"
+
+typedef struct {
+	const char	*name;
+	Type::Code	code;
+} TfwField;
+
+static const TfwField tfw_fields[] = {
+	[TFW_MMAP_LOG_ADDR]		= {"address", Type::IPv6},
+	[TFW_MMAP_LOG_METHOD]		= {"method", Type::UInt8},
+	[TFW_MMAP_LOG_VERSION]		= {"version", Type::UInt8},
+	[TFW_MMAP_LOG_STATUS]		= {"status", Type::UInt16},
+	[TFW_MMAP_LOG_RESP_CONT_LEN]	= {"response_content_length", Type::UInt32},
+	[TFW_MMAP_LOG_RESP_TIME]	= {"response_time", Type::UInt32},
+	[TFW_MMAP_LOG_VHOST]		= {"vhost", Type::String},
+	[TFW_MMAP_LOG_URI]		= {"uri", Type::String},
+	[TFW_MMAP_LOG_REFERER]		= {"referer", Type::String},
+	[TFW_MMAP_LOG_USER_AGENT]	= {"user_agent", Type::String},
+};
+
+TfwClickhouse *tfw_clickhouse;
+TfwColumnFactory *tfw_column_factory;
+
+#ifdef DEBUG
+static void
+hexdump(const char *data, int buflen)
+{
+	const unsigned char *buf = (const unsigned char*)data;
+	int i, j;
+
+#define PRINT_CHAR(c) (isprint(c) ? c : '.')
+	for (i = 0; i < buflen; i += 16) {
+		printf("%06x: ", i);
```

Comment:
it is not good to mix std::cout and printf. May be use `ostringstream`

Reply:
Fixed.